domain/src/main/java/edu/nextstep/camp/calculator/Calculator.kt

```diff
@@ -1,21 +1,17 @@
 package edu.nextstep.camp.calculator
 
+import java.util.*
+
 class Calculator {
 
-    fun evaluate(expression: String?): Int {
-        val expressionStack = ExpressionStack()
-        val expressionWithoutBlank = expression?.split(DELIMITER) ?: throw IllegalArgumentException()
-
-        expressionWithoutBlank.forEach { str ->
-            val stack = expressionStack.getStackForCalculating(str)
-            stack?.let {
-                val type = it.pop()
-                val value = it.pop()
-                val result = calculateValue(type, value.toInt(), str.toInt())
-                expressionStack.pushResult(result.toString())
-            }
-        }
-        return expressionStack.getStackPeekIntegerValue()
+    fun evaluate(stack: Stack<String>): Int {
+        if (stack.size == 1) return stack.pop().toInt()
+        val value = stack.pop().toInt()
+        val type = stack.pop()
+        val value2 = stack.pop().toInt()
+        val result = calculateValue(type, value, value2)
+        stack.push(result.toString())
+        return evaluate(stack)
     }
```

Comment:
evaluate 함수가 Expression의 내부 자료구조인 Stack을 꺼내서 받는건 Expression이 제대로 된 역할을 못해주는것 같습니다.
이전 리뷰에서 ExpressionStack의 이름을 Expression으로 바꾼건 내부 데이터를 감추기 위함입니다. 
Stack을 제어하는건 전적으로 Expression객체에 전적으로 위임하는게 어떨까요?

```suggestion
    fun evaluate(expression: Expression): Int {
        if (!expression.isReadyForCalculating()) {
            return expression.getCurrent()
        }
         
        val operand1 = expression.getOperand1()
        val operator = expression.getOperator()                
        val operand2 = expression.getOperand2()                                
        val result = calculateValue(operator, operand1, operand2)
                
        expression.pushResult(result.toString())
        
        return evaluate(expression)
    }
```

원래 코드를 기반으로 비슷하게 stack 제어를 expression 객체에게 위임한 형태로 작성해보았습니다. 😘